### ISSUE TITLE
Particles with an assigned object - bugs with AsJSON and Set from JSON

### ISSUE THREAD

**Issue by dop2000** [Scirra/Construct-3-bugs#6156]
## Problem description

If a sprite is selected as an object for particles, restoring the particles state from JSON doesn't restore sprites.
Also, if an instance of a sprite is destroyed, trying to access ParticlesJSON expression throws an exception in console log.

## Attach a .c3p

https://www.dropbox.com/s/twx11mbynkd77jb/Particles_issues.c3p?dl=0

## Steps to reproduce

First issue: Try pressing the button multiple times.
 
Second issue: Activate the "Bug2" group, run the project and open console log.

## Observed result

Particles1 object is restored correctly.
Particles2 object restarts from the beginning.

## Expected result

Restoring particles state from JSON is often used to instantly fill the screen with particles. It should also work when a sprite object is assigned.

The second bug is not so important, I just noticed it in the console log.

## System details

<details><summary>View details</summary>

Platform information
Product: Construct 3 r308.2 (stable)
Browser: Chrome 105.0.5195.127
Browser engine: Chromium
Context: webapp
Operating system: Windows 10
Device type: desktop
Device pixel ratio: 1.75
Logical CPU cores: 16
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36
Language setting: en-US

Local storage
Storage quota (approx): 275 gb
Storage usage (approx): 532 mb (0.2%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc. (AMD)
Renderer: ANGLE (AMD, AMD Radeon(TM) Graphics Direct3D11 vs_5_0 ps_5_0, D3D11)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_norm16
KHR_parallel_shader_compile
OES_draw_buffers_indexed
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
OVR_multiview2
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm; codecs=opus)
Ogg Opus (audio/ogg; codecs=opus)
WebM Vorbis (audio/webm; codecs=vorbis)
Ogg Vorbis (audio/ogg; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm; codecs=av01.0.00M.08)
MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.265 (video/mp4; codecs=hev1.1.2.L93.B0)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
This is difficult to investigate for two reasons:

1) You've mentioned two problems in one report, which always confuses things - please make one report per issue and include all the requested information separately for each problem, as per the bug report guidelines.

2) The provided project appears to include big blocks of hard-coded JSON data strings. I don't know how you got this data, what the correct result would be, or how to reproduce the problem if I need to obtain that data string again. I think it would be much easier to deal with if the project just includes both a save and load button and doesn't need to hard-code complex details.

I think this issue is already too confusing and hard to follow to investigate, and would be better off filed as two separate issues, so closing. Please always follow the full bug report guidelines no matter how experienced you are with Construct or how many issues you've filed before. We genuinely need all the details provided and explained clearly as per the bug report guidelines, as without that it basically becomes impossible to investigate.
